### PR TITLE
fix(event-hub): entra disabled by default for single sub onboardings (Internal)

### PR DESCRIPTION
**Pull request by matteopasa**
We will introduce 2 flags, one for enabling entra on single sub, and one for enabling entra on org.
By default org will have Entra enabled while single subs will have Entra disabled by default.

### CODE REVIEW

**Review by nkraemer-sysdig — APPROVED** [review]

File: modules/integrations/event-hub/main.tf [hunk 1]
```diff
@@ -139,7 +139,7 @@ resource "azurerm_monitor_diagnostic_setting" "sysdig_diagnostic_setting" {
 }
 
 resource "azurerm_monitor_aad_diagnostic_setting" "sysdig_entra_diagnostic_setting" {
-  count = var.enable_entra ? 1 : 0
+  count = coalesce(var.enable_entra, var.is_organizational) ? 1 : 0
```
Comment (cgeers) [inline modules/integrations/event-hub/main.tf:142]: This seems like an interesting use of the `coalesce` builtin. I'm curious if other alternatives that were considered.

File: modules/integrations/event-hub/variables.tf [hunk 2]
```diff
@@ -105,9 +105,8 @@ variable "management_group_ids" {
 }
 
 variable "enable_entra" {
-  description = "(Optional) Used to enable or disable Entra logs, defaults to true."
+  description = "(Optional) Used to enable or disable Entra logs."
   type        = bool
-  default     = true
```
Comment (ravinadhruve10) [inline modules/integrations/event-hub/variables.tf:110]: Does this make `enable_entra` a required variable to be passed?

Comment (matteopasa) [reply modules/integrations/event-hub/variables.tf:110]: true, I changed it to have 2 enable entra flags, one for single sub and one for org

File: modules/integrations/event-hub/main.tf (hunk 1 above)
Comment (matteopasa) [reply modules/integrations/event-hub/main.tf]: I actually removed coalesce, because I think it will not work with booleans

**Review by cgeers — APPROVED** [review]
the two similar entra bool variables make it more pedantic than I'd like, but I guess this is necessary to achieve the intended default behavior.

File: modules/integrations/event-hub/variables.tf (hunk 2 above)
Comment (ravinadhruve10) [reply modules/integrations/event-hub/variables.tf:110]: Thanks! :)

**Review by ravinadhruve10 — APPROVED** [review]

**Comment by nkraemer-sysdig** [comment]
@matteopasa this change will invalidate any existing snippets that were using the `enable_entra` variable. Is this present in any hosted snippets/are you okay with breaking any existing snippets using this var in customer envs?

**Comment by ravinadhruve10** [comment]
> @matteopasa this change will invalidate any existing snippets that were using the `enable_entra` variable. Is this present in any hosted snippets/are you okay with breaking any existing snippets using this var in customer envs?

+1. That's a good call out actually.

**Comment by matteopasa** [comment]
Thanks for all the comments @nkraemer-sysdig @cgeers @ravinadhruve10, I agree that we don't want to break existing snippets, closing this and replacing it with disabling entra for single case when the tf is requested: 

- onboarding-api tf: https://github.com/draios/secure-backend/pull/33800
- UI tf: https://github.com/draios/ui/pull/7469

**Pull request closed without merging**